src/Toml/Parser.hs

```diff
@@ -79,12 +80,66 @@ basicStringP :: Parser Text
 basicStringP = lexeme $ Text.pack <$> (char '"' *> anyChar `manyTill` char '"')
 
 textP :: Parser Text
-textP = literalStringP <|> basicStringP
+textP = multilineStringP <|> multilineLiteralStringP <|> literalStringP <|> basicStringP
 
 -- adds " or ' to both sides
 quote :: Text -> Text -> Text
 quote q t = q <> t <> q
 
+nonControlChar :: Parser Text
+nonControlChar = Text.singleton <$> satisfy (not . isControl)
+
+multilineStringP :: Parser Text
+multilineStringP = lexeme $ fmap mconcat $ quotes3
+                                        >> optional eol
+                                        >> allowedChar `manyTill` quotes3
+  where
+    quotes3 = string "\"\"\""
+
+    allowedChar :: Parser Text
+    allowedChar = lineEndingBackslash <|> escapeSequence <|> nonControlChar <|> eol
+
+    lineEndingBackslash :: Parser Text
+    lineEndingBackslash = Text.empty <$ try (char '\\' >> eol >> space)
+
+    escapeSequence :: Parser Text
+    escapeSequence = char '\\' >> anyChar
+                 >>= \case
+                      'b'  -> return "\b"
+                      't'  -> return "\t"
+                      'n'  -> return "\n"
+                      'f'  -> return "\f"
+                      'r'  -> return "\r"
+                      '"'  -> return "\""
+                      '\\' -> return "\\"
+                      'u'  -> parseHexUnicode 4
+                      'U'  -> parseHexUnicode 8
+                      _    -> fail ""
+      where
+        parseHexUnicode :: Int -> Parser Text
+        parseHexUnicode n = count n hexDigitChar
+                        >>= \x -> case (toUnicode . toInt) x of
+                              Just c  -> return (Text.singleton c)
+                              Nothing -> fail ""
+          where
+            toInt xs = (read :: String -> Int) ("0x" ++ xs)
+            toUnicode x
```

Comment:
Looks weird that this function is not anywhere on Haddock 🤔 But I didn't found it either. I can recommend asking on SO.

Reply:
So, there should be a function that works like [`chr`](https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Char.html#v:chr) but fails if its argument is not a Unicode scalar value? I can ask about that.

Reply:
@ghallak Yeah, you could ask about that. Good idea!